src/Sulu/Bundle/MediaBundle/Media/PropertiesProvider/PropertiesProviderInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\Media\PropertiesProvider;
+
+use Symfony\Component\HttpFoundation\File\File;
+
+interface PropertiesProviderInterface
+{
+    /**
+     * @return array<string, mixed>
+     */
+    public function provide(File $file): array;
+
+    public static function supports(File $file): bool;
```

Comment:
do you see an advantage of using a `supports` method? the provider could just return an empty array if he cannot analyze the file right?

Reply:
@nnatter the concept is that it does not initialize the provider service until the service does support it. But did not yet test it if the lazy mechanic works as expected and does not initialize the ffmpeg service when an image is uploaded.

Reply:
okay - not sure if the optimization is worth the more complex interface. but dont have a too strong opinion 🙂